src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
 
+function formatNumber(number: number): number {
+  return Math.floor(number * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a + b <= c || a + c <= b || c + b <= a) {
+      throw new Error(
+        'triangles sides must be less than the sum of the two other sides',
+      );
+    }
 
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
This and similar checks can be made in a helper function that accepts any number of parameters and loops through them using the array iteration method (f.e. some or every).